internetee/auction_center
# New UI 2023

**Tasks:**
- Update auction UI

**New approaches**
- Using view_component, jsbuild, propshaft
- Was added new setting items
  - invoice_issuer_address
  - invoice_issuer_reg_no
  - invoice_issuer_vat_number
  - organization_phone

**Pipeline**
- Make migration
- Precompile assets

**What has been done / needs to be done**
- [x] Updated auction page
- [x] Updated my accounts page
- [x] Updated my rates page
- [x] Implemented modal windows
- [x] Added filtering and search on the auctions page
- [x] Connected the work of web sockets
- [x] Connected automatic calculation of the amount of bets + tax
- [x] Made it possible to bid for the English auction
- [x] Implement autobidder
- [x] Add notifications. I use toast js library to make notifications more beauty according design
- [x] Implement show all button
- [x] Implemented my offers page
- [x] Implemented Tab feature 
- [x] Implemented invoices page
- [x] Implement "Pay all unpaid invoices"
- [ ] Add sort feature for invoices table
- [ ] Pagination for invoices
- [ ] Show all for invoices
- [ ] Show all animation
- [x] Handle modal window in mobile/table view. Maybe it should be as separate page, not a modal
- [x] Implement notification page
- [x] Need to add different icon types for different types of notifications
- [x] Implement read feature for notifications(and read all)
- [x] Implement blind auction bids
- [x] Fix when modal window is open at the empty page without content
- [x] Make columns as sortable in my offers page
- [x] Finish complex sortable column as `total` in offer page
- [ ] Implement endpoint for show all in offers page
- [ ] at the moment to find the result by domain on the results page in the admin panel it is necessary to enter the full domain name, it should be changed so that it searches automatically and starts searching by the entered pattern.
- [x] Add deposit view to auction list
- [x] Configure mobile view (connect javascript actions)
- [x] Block autobider toggle if price is less than minimum bid
- [ ] Implement page of versions
- [x] Add timeleft for english auctions
- [x] Add pagination, make according style
- [x] Made the table as a customizable component
- [x] Add sorting for table
- [x] Fix "Show auctions with no start date" in admin panel. WHen in turn on, it works fine, but not work when rollback
- [x] Implement admin auction page with all neccessary logic
- [x] Make sort table in its own frame 
- [x] Update information about issuer in invoice details. Setting.find_by(code: 'invoice_issuer').retrieve - display all information about issuer in one line, but according by design there are should be different pieces of current information. For example, address, vat, reg number, issuer name should be as independence entities 
- [ ] Some information may be missing, e.g. user has no notifications, accounts, bets. Then we should make some kind of a stopper, e.g. "no invoices"
- [x] Enable captcha
- [ ] Deposit Invoice Show
- [ ] User cache counter for counting
- [x] Configure Turbo Drive
- [x] Move UI logic into view components
- [ ] Get rid from presenters, because now logic in view components
- [ ] Get rid from comments code
- [ ] Get rid from packs directory
- [ ] Get rid from not-used localization
- [ ] Get rid from "dead" code
- [ ] Implement error handle pages - 404 and 500
- [x] Fix webpush toggle value
- [x] When autobidder is set, page is refreshed. It shouldn't be, need to use Turbo features
- [x] Implement History page
- [x] Strange sorting in auction
- [ ] Finish with sorting in order page
- [x] Implement GDPR banner
- [x] Implement timezone browser localization
 ...

**How to test?**
- Check the bids for the English auction
- Check the possibility of calculating the total amount of bets by entering the price of the bet, as well as switching the billing profile
- Check if web sockets work. NB! Make sure websockets don't send prohibited information, for example, users can't see what bids were made for a blind auction
- Check type of notifications. Depends of user is win or lost, should render different messages with differents icons
- Check how admin panel work (filters, set values for auctions, sorting and etc.)
- Check mobile view
- Check password handles(reset password, forget password, input new password, not received password etc.)

**Security check**
- some action restricted by javascript, but what if javascript is turn off?!
- check architecture design security
- websocket security
- Check in page "your offer". Does it really your offer or not? Also check: do web sockets change the behavior of this page?
- Check details for your offers what related to blind auction. There aren't be any of bids from another participants
- Need to put uuid of english auction to the endpoints what related for blind auction. Should be show 404 page
- If somebody another win, need to be sure that in offers another person won't see registration code
- Check recaptcha for blind offers, english offers and autobider
- check sorting, searching, flitering
